tls/s2n_resume.c

```diff
@@ -209,36 +223,45 @@ static S2N_RESULT s2n_tls13_client_deserialize_session_state(struct s2n_connecti
      */
     uint64_t current_time = 0;
     RESULT_GUARD_POSIX(conn->config->wall_clock(conn->config->sys_clock_ctx, &current_time));
-    RESULT_GUARD(s2n_validate_ticket_age(current_time, issue_time));
+    RESULT_GUARD(s2n_validate_ticket_age(current_time, psk.ticket_issue_time));
 
-    uint32_t ticket_age_add = 0;
-    RESULT_GUARD_POSIX(s2n_stuffer_read_uint32(from, &ticket_age_add));
+    RESULT_GUARD_POSIX(s2n_stuffer_read_uint32(from, &psk.ticket_age_add));
 
     uint8_t secret_len = 0;
     RESULT_GUARD_POSIX(s2n_stuffer_read_uint8(from, &secret_len));
     RESULT_ENSURE_LTE(secret_len, S2N_TLS_SECRET_LEN);
-    
-    uint8_t secret[S2N_TLS_SECRET_LEN] = { 0 };
-    RESULT_GUARD_POSIX(s2n_stuffer_read_bytes(from, secret, secret_len));
+    uint8_t *secret_data = s2n_stuffer_raw_read(from, secret_len);
+    RESULT_ENSURE_REF(secret_data);
+    RESULT_GUARD_POSIX(s2n_psk_set_secret(&psk, secret_data, secret_len));
+
+    uint32_t max_early_data_size = 0;
+    RESULT_GUARD_POSIX(s2n_stuffer_read_uint32(from, &max_early_data_size));
```

Comment:
Do we need to worry about compatibility with state serialized prior to this version and thus there will not be any more data available when we reach this call?

Reply:
We don't, because we haven't published these APIs yet, so no TLS1.3 tickets exist in the wild.